Use proper headers to auth with Vault

omgjlk opened the pull request:
##### SUMMARY
Reading examples at
https://learn.hashicorp.com/vault/getting-started/apis show needing to
use `X-Vault-Token` header, instead of `Authorization`. Without this
header, the vault server would return a 400 status with an error message
of "missing client token". With this change AWX is now able to interface
with the Hashicorp backend.

Edited with more context.

It appears that recent versions of HashiCorp Vault support the `Authorization` header in addition to the `X-Vault-Token` header. However older versions (like the one we're using) do not support `Authorization` and only support `X-Vault-Token`. I believe that AWX should default to the header that's most widely supported, until such time as Vault no longer accepts that header.

https://www.vaultproject.io/docs/auth/token.html#via-the-api

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 7.0.0
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

omgjlk committed:
Use proper headers to auth with Vault

Reading examples at
https://learn.hashicorp.com/vault/getting-started/apis show needing to
use `X-Vault-Token` header, instead of `Authorization`. Without this
header, the vault server would return a 400 status with an error message
of "missing client token". With this change AWX is now able to interface
with the Hashicorp backend.

matburt commented:
I don't think we can actually remove the use of Bearer tokens at this point. We might can add this one, but I'm not sure how I feel about this.

omgjlk committed:
Restore new style headers

This leads to having both the new style header and the old compatability
header. Best of both worlds!

omgjlk commented:
> I don't think we can actually remove the use of Bearer tokens at this point. We might can add this one, but I'm not sure how I feel about this.

I've added back the Bearer token. Now both exist.

jakemcdermott commented on awx/main/credential_plugins/hashivault.py:
```diff
@@ -103,6 +103,8 @@ def kv_backend(**kwargs):
 
     sess = requests.Session()
     sess.headers['Authorization'] = 'Bearer {}'.format(token)
+    # Compatability header for older installs of Hashicorp Vault
```
:+1: Thanks. lgtm.
